src/public/js/main.js

```diff
@@ -15,16 +17,33 @@
       }
 
       var msg = $('#post').val();
-      var encoded = RS.encode(msg);
+      var i = 1;
+      var last = 0;
+      var encoded = new Array(0);
```

Comment:
The reason not to is that as far as I can tell there is no way to specify an underlying array length to the constructor. So per-allocating would make an array full of undefineds which must be changed instead of an array we can push new things on to.. It was early in the morning when I wrote this and I was also working so I didn't think it through particularly well and I in fact have a better plan here performance wise(I think) by just assigning the output of RS.encode to the encoded array. I am working on that now.